src/main/java/org/geysermc/discordbot/commands/FuidCommand.java

```diff
@@ -0,0 +1,101 @@
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @author GeyserMC
+ * @link https://github.com/GeyserMC/GeyserDiscordBot
+ */
+
+package org.geysermc.discordbot.commands;
+
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.jagrosh.jdautilities.command.SlashCommand;
+import com.jagrosh.jdautilities.command.SlashCommandEvent;
+
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.interactions.commands.OptionType;
+import net.dv8tion.jda.api.interactions.commands.build.OptionData;
+import org.geysermc.discordbot.util.BotColors;
+import org.geysermc.discordbot.util.HttpUtils;
+
+import java.time.Instant;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+
+public class FuidCommand extends SlashCommand {
+    private static final String urlString = "https://api.geysermc.org/v2/xbox/xuid/";
+
+    public FuidCommand() {
+        this.name = "fuuid";
+        this.help = "get floodgate uuid from player";
+        this.arguments = "<bedrock-username>";
+        this.guildOnly = false;
+
+        this.options = Collections.singletonList(
+                new OptionData(OptionType.STRING, "bedrock-username", "username to grab floodgate uuid").setRequired(true)
+        );
+    }
+
+    @Override
+    protected void execute(SlashCommandEvent event) {
+        String username = Objects.requireNonNull(event.getOption("bedrock-username")).getAsString();
+
+        HttpUtils.asyncGet(urlString + username)
+                .whenComplete((result, error) -> {
+                    if (error != null) {
+                        error.printStackTrace();
+                        return;
+                    }
+
+                    JsonObject response = result.getResponse();
+                    JsonElement xuidElement = response.get("xuid");
+
+                    if (xuidElement == null) {
+                        event.replyEmbeds(handle(username, "User was not found")).queue();
+                        return;
+                    }
+
+                    long xuid = xuidElement.getAsLong();
+                    UUID floodgateUUID = getJavaUuid(xuid);
+                    event.replyEmbeds(handle(username, String.valueOf(floodgateUUID))).queue();
+                });
+    }
+
+    private MessageEmbed handle(String username, String fuuid) {
+        EmbedBuilder builder = new EmbedBuilder()
+                .setTitle("Floodgate UUID")
+                .addField("PlayerName", username, false)
+                .addField("Floodgate UUID", String.valueOf(fuuid), false)
+                .setTimestamp(Instant.now())
+                .setColor(BotColors.SUCCESS.getColor());
+        if (fuuid.contains("User")){
+            builder.setColor(BotColors.FAILURE.getColor());
+        }
+
+        return builder.build();
+    }
+
+    public static UUID getJavaUuid(long xuid) {
```

Comment:
This is called once and does only a simple operation, I suggest just inlining it